### ISSUE TITLE
Provide next page / previous page links in the documentation

### ISSUE THREAD

**Issue by smarter** [lampepfl/dotty#2662]
The reference section of the documentation is almost like a small book on Dotty, as such it would be nice if it could be navigated more like an online book, either with links at the bottom, or with giant arrows around the page like in https://doc.rust-lang.org/beta/book/second-edition/

**Comment by felixmulder** [comment]
I like this suggestion - however it would take us further from being a drop-in replacement for Jekyll. This would require us to implement some form of indexing of the different pages. Maybe the enumeration could be done automatically,  but how do you then control in which order sections come?

Should we hardcode it in each page? That doesn't really scale to other projects since they'd have to do the same.

Should we implement some form of functionality to indicate which the next page is? And if so - that should be done in a way that uses the preamble.

I don't think this is a clear cut issue yet - but I'm happy to instigate conversation and move this forward :)

**Comment by smarter** [comment]
Good point. I think having some way to order the pages would be good, this way we could also automatically generate the sidebar. The simplest way I can think of is to prefix the md files:, eg.: `docs/02_reference/03_changed/01_enums.md`.
